### PR TITLE
[Crane] Initalises the map only once

### PR DESCRIPTION
**Pull request by manuelvicnt**

### CODE REVIEW

**Review by chrisbanes — APPROVED** [review]

File: Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt [hunk 1]
```diff
@@ -153,7 +153,8 @@ private fun MapViewContainer(
     latitude: String,
     longitude: String
 ) {
-    var zoom by rememberSaveable { mutableStateOf(InitialZoom) }
+    var zoom by rememberSaveable(map) { mutableStateOf(InitialZoom) }
+    var mapInitialized by rememberSaveable(map) { mutableStateOf(false) }
```
Comment (chrisbanes) [inline Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt:157]: This should just be `remember` right? You'll need to re-initialize after activity death

File: Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt [hunk 2]
```diff
@@ -163,14 +164,20 @@ private fun MapViewContainer(
         // Reading zoom so that AndroidView recomposes when it changes. The getMapAsync lambda
         // is stored for later, Compose doesn't recognize state reads
         val mapZoom = zoom
+        if (!mapInitialized) {
```
Comment (chrisbanes) [inline Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt:167]: Probably better done in a launched effect?

```
LaunchedEffect(mapInitialized) {
    if (!mapInitialized) {
        // blah
    }
}
```

File: Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt (hunk 1 above)
Comment (manuelvicnt) [reply Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt]: Sure, done! Thanks

File: Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt (hunk 2 above)
Comment (manuelvicnt) [reply Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt]: Yeah! Much better, thanks!

File: Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt [hunk 3]
```diff
@@ -153,24 +153,32 @@ private fun MapViewContainer(
...
+            val position = LatLng(latitude.toDouble(), longitude.toDouble())
+            googleMap.addMarker {
+                position(position)
+            }
+            googleMap.moveCamera(CameraUpdateFactory.newLatLng(position))
+            mapInitialized = true
+        }
+    }
 
+    var zoom by remember(map) { mutableStateOf(InitialZoom) }
```
Comment (chrisbanes) [inline Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt:169]: This should probably still be `rememberSaveable`?

Comment (manuelvicnt) [reply Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt]: Makes sense. Thanks!

**Review by chrisbanes — APPROVED** [review]
🚢

**Pull request merged**